File: RACApp2/RACApp2/ViewController.m
//
//  ViewController.m
//  RACApp2
//
//  Created by Raunak Talwar on 8/14/15.
//  Copyright (c) 2015 Raunak Talwar. All rights reserved.
//

#import "ViewController.h"
#import <ReactiveCocoa/ReactiveCocoa.h>
@interface ViewController ()<UITextFieldDelegate>
@property (strong, nonatomic) IBOutlet UITextField *username;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view,

    NSArray *numbers = @ [ @1, @2, @3];
    RACSequence *sequence = [ numbers.rac_sequence map:^id(NSNumber *number) {
        return @([number intValue] * [number intValue]);
    }];
    NSLog(@"%@",sequence.head);
    
    //Using RACObserve
    
    [self.username becomeFirstResponder];
    [RACObserve(self, username.text)
     subscribeNext:^(NSString *name) {
        NSLog(@"%@", name);
    }];


    self.username.text = @"Hi";

    
    
    
    
}
- (IBAction)doneButtonClicked:(id)sender {
    NSLog(@"self.textField.text =%@", self.username.text);

}

- (void)textFieldDidBeginEditing:(UITextField *)textField
{
    NSLog(@"textFieldDidBeginEditing");
    
}

- (IBAction)touchUpInside:(id)sender {
    NSLog(@"self.touchupinside");
}

- (IBAction)valueChanged:(id)sender {
    NSLog(@"self.textField.text =%@", self.username.text);
}

- (BOOL)prefersStatusBarHidden
{
    return YES;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift-Learn-Project-1/Swift-Learn-Project-1/ViewController.swift
//
//  ViewController.swift
//  Swift-Learn-Project-1
//
//  Created by Raunak Talwar on 9/1/15.
//  Copyright (c) 2015 Raunak Talwar. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var outputLabel: UILabel!
    var counter:Int = 0
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    @IBAction func buttonPressed(sender: AnyObject) {
        counter = counter + 1;
        outputLabel.text = "Button Pressed \(counter) times";
        
    
    }
}

